### ISSUE TITLE
JSONNull is being emitted for optional values

### ISSUE THREAD

**Issue by dvdsgl** [quicktype/quicktype#365]
```swift
// To parse the JSON, add this file to your project and do:
//
//   let coordinate = try? JSONDecoder().decode(Coordinate.self, from: jsonData)

import Foundation

struct Coordinate: Codable {
    let latitude, longitude: Double?
}

// MARK: Encode/decode helpers

class JSONNull: Codable {
    public init() {}

    public required init(from decoder: Decoder) throws {
        let container = try decoder.singleValueContainer()
        if !container.decodeNil() {
            throw DecodingError.typeMismatch(JSONNull.self, DecodingError.Context(codingPath: decoder.codingPath, debugDescription: "Wrong type for JSONNull"))
        }
    }

    public func encode(to encoder: Encoder) throws {
        var container = encoder.singleValueContainer()
        try container.encodeNil()
    }
}
```

**Comment by dvdsgl** [comment]
Wow! We found a GitHub bug (that this issue was closes by the PR).

**Comment by schani** [comment]
I'm guessing they're only searching for a string like "Fix(es)? #9999", and don't care whether it's in an unchecked box.

**Comment by lutluthfi** [comment]
Could you guys provide a documentation how to use `JSONNull` class, please?
